### PR TITLE
Add requirePasswordChange with a default of true to all procedures that enable access for a user

### PR DESCRIPTION
**Pull request by boggle**
Had to re-open due to git foo (was: 7812)

### CODE REVIEW

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/NeoShallowEmbeddedInteraction.java [hunk 1]
```diff
@@ -35,6 +35,7 @@
 import org.neo4j.test.TestEnterpriseGraphDatabaseFactory;
 
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
```
Comment (fickludd) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/NeoShallowEmbeddedInteraction.java]: This import is redundant, or rather poor ascii-art.

File: enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java [hunk 2]
```diff
@@ -57,30 +57,37 @@
...
+            @Name( "newPassword" ) String newPassword,
+            @Name( value = "requirePasswordChange", defaultValue = "true" ) boolean requirePasswordChange
+    )
             throws InvalidArgumentsException, IOException
     {
         EnterpriseAuthSubject enterpriseSubject = EnterpriseAuthSubject.castOrFail( authSubject );
         if ( enterpriseSubject.doesUsernameMatch( username ) )
         {
-            enterpriseSubject.setPassword( newPassword );
+            enterpriseSubject.setPassword( newPassword, false );
```
Comment (craigtaverner) [inline enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java]: If the user passes a true here, should we not use that, or alternatively throw an exception if we think it too irrational to set your own password with a require password change? I think we should just do what the user says. I can imagine a user wanting to do this.

File: community/security/src/main/java/org/neo4j/server/security/auth/AuthProcedures.java [hunk 3]
```diff
@@ -40,7 +40,7 @@
     public AuthSubject authSubject;
 
     @Description( "Create a user." )
-    @Procedure( name = "dbms.createUser", mode = DBMS )
+    @Procedure( name = "dbms.security.createUser", mode = DBMS )
     public void createUser( @Name( "username" ) String username, @Name( "password" ) String password,
             @Name( "requirePasswordChange" ) boolean requirePasswordChange )
```
Comment (craigtaverner) [inline community/security/src/main/java/org/neo4j/server/security/auth/AuthProcedures.java]: The community version seems to be missing the default value. Should be the same signature and default as enterprise.

File: community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresIT.java [hunk 4]
```diff
@@ -92,7 +92,7 @@ public void shouldNotChangeOwnPasswordIfNewPasswordInvalid() throws Exception
     @Test
     public void shouldCreateUser() throws Exception
     {
-        assertEmpty( admin, "CALL dbms.createUser('andres', '123', true)" );
+        assertEmpty( admin, "CALL dbms.security.createUser('andres', '123', true)" );
```
Comment (craigtaverner) [inline community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresIT.java:95]: Seem to be missing tests in community for creating user with default value of changePasswordRequired

File: enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java (hunk 2 above)
Comment (fickludd) [reply enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java]: No way to find out whether a `true` is default or explicit at this point right? I think it's fair game to use the actual `requiredPasswordChange` then. If you want to change your own password in a smooth way, you would use `changePassword` instead.

File: community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresIT.java (hunk 4 above)
Comment (OliviaYtterbrink) [inline community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresIT.java]: added tests

Comment (OliviaYtterbrink) [reply community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresIT.java:95]: Added tests

**Pull request merged**